### ISSUE TITLE
deleteFile(id) does not return callback/promise/ or correct boolean

### ISSUE THREAD

**Issue by TimDaub** [FineUploader/fine-uploader#1426]
Hi,

I'm trying to delete a file on the server and I'm using `deleteFile(id)` (http://docs.fineuploader.com/branch/master/api/methods.html#deleteFile).

I have `CORS` enabled, but didn't yet implement it on the server and so when I call the method I get:

```
XMLHttpRequest cannot load http://<myURL>/s3/delete/<MyFileName>?key=<MyKey>&bucket=<MyBucket>. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8000' is therefore not allowed access.
s3.fine-uploader.js:243 [Fine Uploader 5.2.2] DELETE request for 0 has failed - response code 0
s3.fine-uploader.js:243 [Fine Uploader 5.2.2] Delete request for '3z7FbQa (3rd copy).png' has failed.
```

Still, `deleteFile(id)`, returns `true` everytime, even though the request failed.
(Btw.: The documentation says it returns nothing...)

But all that aside: Wouldn't it be beneficial if deleteFile(id) would return a promise, so that I as a developer can delete the file from my view when it is actually successfully deleted on the server?

**Comment by rnicholus** [comment]
We can look into explicitly returning a status value from this method, specifically a promise, in some future release.

**Comment by TimDaub** [comment]
Yeah, that sounds good, though it would be even cooler if fineUploader
would either use es6-promises (as a polyfill) or q(Promises A+).

Because right now, this adds more payload to our application, as we don't
use fineUploader's promises anywhere except for in fineuploader specific
code.

**Comment by rnicholus** [comment]
The internal promise impl probably accounts for about 500 bytes gzipped. Hardly noticeable.

**Comment by TimDaub** [comment]
Ok, fair enough but without looking into it, is it compliant with:
https://gist.github.com/domenic/3889970 or the Promise A+ spec?

**Comment by rnicholus** [comment]
No, it is not compliant, but more than sufficient for the purposes here. This is outlined in the promise module docs. For more info please read http://docs.fineuploader.com/branch/master/features/async-tasks-and-promises.html
